testutils/vcamera/vcamera.go

```diff
@@ -0,0 +1,223 @@
+//go:build linux
+
+/*
+Package vcamera creates and streams video to virtual V4L2 capture devices on Linux.
+
+It uses [V4L2Loopback] to create virtual camera drivers and [GStreamer] to display test video streams. You can install
+both using our V4L2Loopback setup [script]. This script also needs to run `sudo modprobe v4l2loopback`. Since
+feeding the root password to that command at runtime would be impractical, it's recommended that you allow the current user
+to run that command without requiring a password. The script above will display a prompt asking if you want this behavior.
+Select "yes".
+
+Usage:
+
+	// create a builder object
+	config := vcamera.Builder()
+
+	// create 1-to-N cameras
+	config = config.NewCamera(1, "Low-res Camera", vcamera.Resolution{Width: 640, Height: 480})
+	config = config.NewCamera(2, "Hi-res Camera", vcamera.Resolution{Width: 1280, Height: 720})
+	...
+
+	// start streaming
+	config, err := Stream()
+	if err != nil {
+		// handle error
+	}
+
+	// shutdown streams
+	config.Shutdown()
+
+Because this class allows for method chaining the above could be accomplished like so
+
+	config, err := vcamera.Builder().
+		NewCamera(1, "Low-res Camera", vcamera.Resolution{Width: 640, Height: 480}).
+		NewCamera(2, "Hi-res Camera", vcamera.Resolution{Width: 1280, Height: 720}).
+		Stream()
+
+	// DO NOT forget to stop streaming to avoid leaking resources.
+	defer config.Shutdown()
+
+[V4L2Loopback]: https://github.com/umlaeute/v4l2loopback/tree/8cb5270d20484bde26eabb085011a8ea27285446
+[GStreamer]: https://gstreamer.freedesktop.org/
+[script]: https://github.com/viamrobotics/rdk/blob/0f49904ec550e5a33fc649e3153b4d465256a02a/etc/v4l2loopback_setup.sh
+*/
+package vcamera
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"os/exec"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/edaniels/golog"
+	"github.com/pkg/errors"
+	"go.viam.com/utils"
+)
+
+// Resolution stores the Width and Height in pixels for a camera resolution.
+type Resolution struct {
+	Width  int
+	Height int
+}
+
+// Config is a builder object used to create virtual cameras.
+type Config struct {
+	devices                 map[int]bool
+	deviceKeys              []string
+	labels                  []string
+	resolutions             []Resolution
```

Comment:
Might be cleaner to make a data structure for Device instead of having arrays for attributes in Config.

```
type Device struct {
    key string
    label string
    resolution Resolution
}
```

Not sure - just throwing it out there.
type Config strcut {
    devices []Device
    ...
}
```

Reply:
Yeah, that's a good idea. Done.